garage/tf/algos/npo.py

```diff
@@ -22,59 +20,101 @@
 from garage.tf.optimizers import LbfgsOptimizer
 
 
-@unique
-class PGLoss(Enum):
-    VANILLA = 'vanilla'
-    SURROGATE = 'surrogate'
-    SURROGATE_CLIP = 'surrogate_clip'
-
-
 class NPO(BatchPolopt):
     """
     Natural Policy Gradient Optimization.
 
-    Attributes:
-        name(str): The name of the algorithm.
-        lr_clip_range(float): The limit on the likelihood ratio between
-            policies, as in PPO.
-        max_kl_step(float): The maximum KL divergence between old and new
-            policies, as in TRPO.
-        policy_ent_coeff(float): The coefficient of the policy entropy.
-        optimizer(float): The optimizer of the algorithm.
+    Args:
+        pg_loss (str, optional): A string from: 'vanilla', 'surrogate',
+            'surrogate_clip'. The type of loss functions to use.
+        lr_clip_range (float, optional): The limit on the likelihood ratio
+            between policies, as in PPO.
+        max_kl_step (float, optional): The maximum KL divergence between old
+            and new policies, as in TRPO.
+        optimizer (object, optional): The optimizer of the algorithm. Should
+            be the optimizers in garage.tf.optimizers.
+        optimizer_args (dict, optional): The arguments of the optimizer.
+        name (str, optional): The name of the algorithm.
+        policy_ent_coeff (float, optional): The coefficient of the policy
+            entropy. Setting it to zero would mean no entropy regularization.
+        entropy_method (str, optional): A string from: 'max', 'regularized',
+            'no_entropy'. The type of entropy method to use. 'max' adds the
+            dense entropy to the reward for each time step. 'regularized' adds
+            the mean entropy to the surrogate objective. See
+            https://arxiv.org/abs/1805.00909 for more details.
+        use_neg_logli_entropy (bool, optional): Whether to estimate the
+            entropy as the negative log likelihood of the action.
+        use_softplus_entropy (bool, optional): Whether to estimate the softmax
+            distribution of the entropy to prevent the entropy from being
+            negative.
+        stop_entropy_gradient (bool, optional): Whether to stop the entropy
+            gradient.
+
+    Note:
+        sane defaults for entropy configuration:
```

Comment:
can you add some more explicit info on the combinations of entropy arguments which are valid? even i find it a little confusing.

Reply:
Could you tell me what information you think is missing here?

Reply:
sorry, this seems sufficient to me.